models/ai85net_adi_tinierssd.py

```diff
@@ -0,0 +1,461 @@
+        conv12_2_feats = self.aux_convs(fire10_feats)
+
+        # Run prediction convolutions (predict offsets w.r.t prior-boxes and classes in each
+        # resulting localization box)
+        locs, classes_scores = self.pred_convs(fire4_feats, fire8_feats, fire9_feats, fire10_feats,
+                                               conv12_2_feats)
+
+        return locs, classes_scores
+
+    @staticmethod
+    def create_prior_boxes(aspect_ratios=default_aspect_ratios):
+        """
+        Create the prior (default) boxes
+        :return: prior boxes in center-size coordinates
+        """
+
+        fmap_dims = {'fire8': 18,
+                     'fire9': 9,
+                     'fire10': 4,
+                     'conv12_2': 2}
+
+        fmaps = list(fmap_dims.keys())
+
+        obj_scales = {'fire8': 0.15,
+                      'fire9': 0.35,
+                      'fire10': 0.55,
+                      'conv12_2': 0.725}
+
+        if len(aspect_ratios) != len(fmaps):
+            raise ValueError(f'aspect_ratios list should have length {len(fmaps)}')
+
+        if True in (len(aspect_ratios_list) !=
+                    len(TinierSSD.default_aspect_ratios[0])
+                    for aspect_ratios_list in aspect_ratios):
+            raise ValueError(f'Each aspect_ratios list should have length \
+                               {len(TinierSSD.default_aspect_ratios[0])}')
+
+        aspect_ratios = {'fire8': aspect_ratios[0],
+                         'fire9': aspect_ratios[1],
+                         'fire10': aspect_ratios[2],
+                         'conv12_2': aspect_ratios[3]}
+
+        prior_boxes = []
+
+        for k, fmap in enumerate(fmaps):
+            for i in range(fmap_dims[fmap]):
+                for j in range(fmap_dims[fmap]):
+                    cx = (j + 0.5) / fmap_dims[fmap]
+                    cy = (i + 0.5) / fmap_dims[fmap]
+
+                    for ratio in aspect_ratios[fmap]:
+                        prior_boxes.append([cx, cy, obj_scales[fmap] * sqrt(ratio),
+                                            obj_scales[fmap] / sqrt(ratio)])
+
+                        # For an aspect ratio of 1, use an additional prior whose scale is the
+                        # geometric mean of the
+                        # scale of the current feature map and the scale of the next feature map
+                        if ratio == 1.:
+                            try:
+                                additional_scale = sqrt(obj_scales[fmap] *
+                                                        obj_scales[fmaps[k + 1]])
+                            # For the last feature map, there is no "next" feature map
+                            except IndexError:
+                                additional_scale = 1.
+                            prior_boxes.append([cx, cy, additional_scale, additional_scale])
+
+        prior_boxes = torch.FloatTensor(prior_boxes).to(device)  # (1700, 4)

```

Comment:
You may have to pass in `device` or a tensor on that device to get the device from, or do the conversion on the return value of this function instead.

Reply:
Done and commit will be amended